src/vos/vos_pool.c

```diff
@@ -752,6 +757,12 @@ vos_pool_open(const char *path, uuid_t uuid, unsigned int flags,
 		rc = errno;
 		D_ERROR("Error in opening the pool "DF_UUID": %s\n",
 			DP_UUID(uuid), pmemobj_errormsg());
+		/*
+		 * If we get EAGAIN, assume it is from a pthread_key_create
+		 * call made by pmemobj_create. See DAOS-6505.
```

Comment:
May not be worth the detail - possibly change comment to state "made by pmemobj_open"? This is an assumption on my part though - just spent a little while trying unsuccessfully to look through libpmemobj sources to find how it may, from pmemobj_open, finally arrive at os_tls_key_create() that calls the pthread_key_create() function.

Reply:
@kccain, I did mean pmemobj_create:

    pmemobj_create
        pmemobj_createU
            obj_runtime_init
                obj_runtime_init_common
                    lane_recover_and_section_boot
                        pmalloc_boot
                            palloc_boot
                                heap_boot
                                    arena_thread_assignment_init

Reply:
I was thinking the comment should say pmemob_open() because that is the function called by vos_pool_open() invoked just above this check. 

Looking at the stack thanks to your help above for the pmemobj_create flow, perhaps pmemobj_open() gets to os_tls_key_creaste() in a similar way to the stack you listed
pmemobj_open
    pmemobj_openU
        obj_open_common
            obj_runtime_init
                obj_runtime_init_common
                    lane_recover_and_section_boot
                        pmalloc_boot
                            palloc_boot
                                heap_boot
                                    arena_thread_assignment_init

Reply:
I see. That's right---I should have changed the wording to "pmemobj_open".